src/provisioning/api.md

```diff
@@ -0,0 +1,292 @@
+
+
+### Request body
+```json5
+{
+    // ... connection specific details, can be configured.
+}
+```
+### Response
+
+```json5
+{
+    "type": "JiraProject", // The name of the connection
+    "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type in the connection
+    "id": "opaque-unique-id", // An opaque ID used to refer to this connection. Should **NOT** be assumed to be stable.
+    "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+    "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}
+```
+
+## PATCH /v1/{roomId}/connections/{id}
+
+Update a connection's configuration. The `id` refers to the `id` returned in the GET response.
+
+The body of the request is the configuration for the connection, which will be the "ConnectionState" interface for each connection.
+
+### Request body
+```json5
+{
+    // ... connection specific details, can be configured.
+}
+```
+### Response
+
+```json5
+{
+
+    "type": "JiraProject", // The name of the connection
+    "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type in the connection
+    "id": "opaque-unique-id", // An opaque ID used to refer to this connection. Should **NOT** be assumed to be stable.
+    "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+    "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}
+```
+
+## DELETE /v1/{roomId}/connections/{id}
+
+Delete a connection. The `id` refers to the `id` returned in the GET response.
+### Response
+
+```json5
+{
+    "ok": true
+}
+```
+
+# Service specific APIs
+
+Some services have specific APIs for additional functionality, like OAuth.
+
+## GitHub
+
+
+### GET /v1/github/oauth?userId={userId}
+
+
+Request an OAuth url for the given user. Once the user has completed the steps in the OAuth process,
+the bridge will be granted access.
+
+### Response
+
+```json5
+[{
+    "user_url": "https://github.com/login/oauth/authorize?...",
+    "org_url": "https://github.com/apps/matrix-bridge/installations/new",
+}]
+```
+
+### GET /v1/github/account?userId={userId}
+
+Request the status of the users account. This will return a `loggedIn` value to determine if the
+bridge has a GitHub identity stored for the user, and any organisations they have access to.
+
+### Response
+
+```json5
+{
+    "loggedIn": true,
+    "organisations":[{
+        "name": "half-shot",
+        "avatarUrl": "https://avatars.githubusercontent.com/u/8418310?v=4"
+    }]
```

Comment:
not really sure what's going on with the two oauth URLs, but I think the bridge needs to probably request more oauth permissions for the user URL and then when returning organizations/repos filter out the ones that the org auth can't handle. 

UX would be something like:
```
Before we can bridge your repositories, please allow the bridge to interact with github as you:

[Authorize]
```

```
[Click here] to add the bridge to your organization. 

Organization: [ > list] [Refresh]
Repository: [ > list]
[Bridge]
```

Reply:
For clarity: I've finally narrowed this down to my usage of the org-specific and account-specific endpoints. Ignoring `organisations` and the `/orgs/:orgId/repositories` endpoint leaves just `/repositories` which works as expected. 

I did also have some switching logic to handle the two now-unused-by-me endpoints, but as implied I don't really need it anymore. It was to handle the special case of the user's own account not being counted as an org:
![image](https://user-images.githubusercontent.com/1190097/144935275-79e6b1ed-da59-41a0-ac24-57280d6e64b6.png)
![image](https://user-images.githubusercontent.com/1190097/144935321-402f7762-ffe3-4ac2-8eb3-5658f4a30af3.png)